gapis/api/transform/transform_chain.go

```diff
@@ -239,3 +257,31 @@ func mutateAndWrite(ctx context.Context, id CommandID, cmds []api.Cmd, out Write
 
 	return nil
 }
+
+type commandCaptureWriter struct {
+	out  Writer
+	cmds []api.Cmd
+}
+
+func createCommandCaptureWriter(out Writer) *commandCaptureWriter {
+	writer := commandCaptureWriter{
+		out:  out,
+		cmds: make([]api.Cmd, 0),
```

Comment:
simply don't initialize `cmds` here. The zero value for slices is the empty slice.